internal/resource/utils/util.go

```diff
@@ -147,6 +192,90 @@ func GetJob(c client.Client, namespace string, jobName string) (*batchv1.Job, er
 	return job, err
 }
 
+func randomStr() string {
+	parts := strings.Split(uuid.New().String(), "-")
```

Comment:
There is a convenient `rand.String()` method in "k8s.io/apimachinery/pkg/util/rand" which is much more easier to use.

Reply:
`uuid.New()` may introduce some computing overhead